src/validation.cpp

```diff
@@ -4614,13 +4628,13 @@ void Chainstate::LoadExternalBlockFile(
                     // called by concurrent network message processing. but, that is not
                     // reliable for the purpose of pruning while importing.
                     BlockValidationState state;
-                    if (!ActivateBestChain(state, pblock)) {
+                    if (!ActiveChainstate().ActivateBestChain(state, pblock)) {
                         LogPrint(BCLog::REINDEX, "failed to activate chain (%s)\n", state.ToString());
                         break;
                     }
                 }
 
-                NotifyHeaderTip(*this);
+                NotifyHeaderTip(ActiveChainstate());
```

Comment:
Note: it seems like we can default to the active chainstate here because the underlying notification that this sends (`KernelNotifications::headerTip()`) isn't really particular to any chainstate.

Reply:
In commit "Move block-storage-related logic to ChainstateManager" (034f920d1755cff752d46b089ad37bdd703bf896)

It seems like `NotifyHeaderTip` is a "block-storage-related" function just like other functions that are moving from `Chainstate` to `ChainstateManager` so it would be cleaner if  `NotifyHeaderTip` took a `ChainstateManager` argument instead of a `Chainstate` one, and if it called `ActiveChainstate()` internally so `NotifyHeaderTip` callers don't have to consider how the UI reflects background validation.